plot/plot.app.js

```diff
@@ -1157,16 +1285,65 @@
                     if (newValue) vm.groups = newValue;
                 });
             }])
-            .directive('plot', ['$rootScope', '$timeout', function ($rootScope, $timeout) {
-
+            .directive('plot', ['$rootScope', '$timeout','ConfigUtils', function ($rootScope, $timeout,ConfigUtils) {
+                
                 return {
                     link: function (scope, element) {
                         scope.$watch('plots', function (plots) {
                             if (plots) {
                                 $timeout(function() {
                                     for (var i = 0; i < plots.length; i++) {
                                         if (plots[i].id == element[0].attributes['plot-id'].nodeValue) {
-                                            Plotly.newPlot(element[0], plots[i].plot_values.data, plots[i].plot_values.layout, plots[i].plot_values.config).then(function () {
+                                            Plotly.newPlot(element[0], plots[i].plot_values.data, plots[i].plot_values.layout, plots[i].plot_values.config)
+                                            .then(
+                                                element[0].on('plotly_click',function(data){
+                                                    var contextUrlParams=ConfigUtils.getContextHeaderParams();
+
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("y_col_hyperlink_pattern")){
+                                                        var idx=data.points[0].data.x.indexOf(data.points[0].x.toString())
+                                                        if(data.points[0].data.y_col_hyperlink_pattern[idx]!=false && data.points[0].data.y_col_hyperlink_pattern[idx]!=undefined){
+                                                            var qCharacter = data.points[0].data.y_col_hyperlink_pattern[idx].indexOf("?") !== -1 ? "&" : "?";
+                                                            var url=data.points[0].data.y_col_hyperlink_pattern[idx]+qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid;
+                                                            window.open(url,'_blank');
+                                                        }
+                                                    }
+                                                    if(data.hasOwnProperty("points") && data.points[0].data.hasOwnProperty("graphic_link_pattern")){
+                                                        var idx=data.points[0].data.labels.indexOf(data.points[0].label.toString());
```

Comment:
Can you add comments in the code explaining what the index is here and maybe describe what the structure of the `data` object is?

From reading your code, it seems like `data.point[0].data.labels` is the array of labels for the chart and `data.points[0].label` is the label that we are currently looking at.